### ISSUE TITLE
[heft] [typescript plugin] Files in `staticAssetsToCopy` are always copied to `lib` but not the one specified in `tsconfig.json`

### ISSUE THREAD

**Issue by xc2** [microsoft/rushstack#2513]
## Summary

Files in `staticAssetsToCopy` are always copied to `lib` but not the one specified in `tsconfig.json`.

## Repro steps

Here's the repo repros this issue: [https://github.com/xc2/GithubIssues/tree/main/heft-issue-staticAssetsToCopy](https://github.com/xc2/GithubIssues/tree/main/heft-issue-staticAssetsToCopy)

```
# assign `outDir` to `dist` in `tsconfig.json`, and then
pnpm build
```

**Expected result:** `hello.js` and `data.json` in `dist` directory.
**Actual result:** `hello.js` in `dist`, and `data.json` in `lib`

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@rushstack/heft` version? | 0.24.2 |
| Operating system? | Mac |
| Would you consider contributing a PR? | Yes/No |
| Node.js version (`node -v`)? | 14.15.5 |

**Comment by iclanton** [comment]
Good catch. Fixing that here: https://github.com/microsoft/rushstack/pull/2521

**Comment by iclanton** [comment]
This has been published in Heft 0.24.3.
